src/python/pants/engine/goal_test.py

```diff
@@ -1,14 +1,40 @@
 # Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
-from pants.engine.goal import Outputting
-from pants.testutil.engine.util import MockConsole
+from unittest.mock import Mock
 
+import pytest
 
-def test_outputting_goal():
-  class DummyGoal(Outputting):
-    pass
+from pants.engine.console import Console
+from pants.engine.goal import Goal, GoalSubsystem, LineOriented
+from pants.engine.rules import goal_rule
+from pants.testutil.engine.util import MockConsole, run_rule
 
-  console = MockConsole()
-  with DummyGoal.output(None, console) as output:
-    pass
+
+@pytest.mark.skip(
+  reason="Figure out how to create a GoalSubsystem for tests. We can't call global_instance()"
+)
```

Comment:
I have no idea how to get this test working. `Mock()` doesn't work and should also not be accepted as the main functionality we're testing is `GoalSubsystem` and we should test the real thing, not a `Mock`.

Normally, we'd call `global_instance() `and `init_subsystem()`. We can't do that here because `GoalSubsystem` is not a `Subsystem`, but an `Optionable`.

Any ideas?

--

At a minimum, this gets us closer to what the test should look like, at least? 😬